### PR TITLE
Grafana implementation

### PR DESCRIPTION
**Pull request by iw-gtadam**

### CODE REVIEW

File: dns.tf [hunk 1]
```diff
@@ -42,6 +42,21 @@ resource "aws_route53_record" "thanos_loadbalancer" {
...
+  type    = "A"
+  zone_id = data.terraform_remote_state.management_dns.outputs.dataworks_zone.id
+
+  alias {
+    evaluate_target_health = false
+    name                   = aws_lb.monitoring[0].dns_name
+    zone_id                = aws_lb.monitoring[0].zone_id
+  }
+
+  provider = aws.management_dns
```
Comment (dan-hill2802) [inline dns.tf:57]: Can you put the provider as the first line in the resource? Just clearer and more consistent

File: ecs_grafana.tf [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    "portMappings": [
+      {
+        "containerPort": 3000,
+        "hostPort": 3000
+      }
+    ],
+    "logConfiguration": {
+      "logDriver": "awslogs",
+      "options": {
+        "awslogs-group": "${data.terraform_remote_state.management.outputs.ecs_cluster_main_log_group.name}",
```
Comment (dan-hill2802) [inline ecs_grafana.tf:29]: I think it needs it's own log group, rather than using the main cluster one

File: ecs_prometheus.tf [hunk 3]
```diff
@@ -244,3 +254,23 @@ resource "aws_security_group_rule" "allow_ingress_thanos_grpc" {
...
+  count             = length(local.roles)
+  type              = "ingress"
+  to_port           = 3000
+  protocol          = "tcp"
+  from_port         = 3000
+  security_group_id = aws_security_group.prometheus[count.index].id
+  cidr_blocks       = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "allow_egress_thanos_grpc" {
```
Comment (dan-hill2802) [inline ecs_prometheus.tf:268]: What's grpc?

File: dns.tf (hunk 1 above)
Comment (iw-gtadam) [reply dns.tf]: resolved

File: ecs_prometheus.tf (hunk 3 above)
Comment (iw-gtadam) [reply ecs_prometheus.tf:268]: https://thanos.io/quick-tutorial.md/#sidecar-components-sidecar-md

File: ecs_grafana.tf (hunk 2 above)
Comment (iw-gtadam) [reply ecs_grafana.tf:29]: It does, I need to do this for all the containers. So i've created a ticket DW-4386 that will address this.

File: ecs_grafana.tf [hunk 4]
```diff
@@ -0,0 +1,93 @@
+resource "aws_ecs_task_definition" "grafana" {
+  count                    = local.is_management_env ? 1 : 0
+  family                   = "grafana"
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = "512"
+  memory                   = "4096"
```
Comment (ChrisScottThomas) [inline ecs_grafana.tf:7]: Don't need to do this now, but we should repeat the variable values here, so we have consistency.

File: ecs_grafana.tf [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    "cpu": ${var.fargate_cpu},
+    "image": "${data.terraform_remote_state.management.outputs.ecr_grafana_url}",
+    "memory": ${var.fargate_memory},
+    "name": "grafana",
+    "networkMode": "awsvpc",
+    "user": "0:0",
+    "portMappings": [
+      {
+        "containerPort": 3000,
+        "hostPort": 3000
```
Comment (ChrisScottThomas) [inline ecs_grafana.tf:23]: Same for ports.

**Pull request merged**